dateutil/rrule.py

```diff
@@ -658,6 +659,23 @@ def __init__(self, freq, dtstart=None,
             self._bysecond = tuple(sorted(self._bysecond))
             self._original_rule['bysecond'] = self._bysecond
 
+        # rscale and optional skip values
+        if rscale is not None:
+            assert rscale == 'GREGORIAN', 'unsupported RSCALE'
```

Comment:
First off, I'm not comfortable with using assertions for error handling. This should raise an error of some sort unless you're just asserting some invariant of the code (consider that assertions are turned off if Python is run with `python -O`.

I'm of two minds about how to implement this. One thing we could do is have a fixed list of calendar types that are considered valid, even if unsupported, adding a class attribute like this:

```python
# List taken from https://www.unicode.org/repos/cldr/tags/latest/common/bcp47/calendar.xml
VALID_RSCALES = {'GREGORIAN', 'GREGORY',
                 'BUDDHIST', 'CHINESE', 'COPTIC', 'DANGI', ... }
```

Then change this check to:
```python
if rscale is not None:
    rscale = rscale.upper()
    if rscale in self.VALID_RSCALES:
        if rscale not in {'GREGORY', 'GREGORIAN'}:
            raise NotImplementedError('Non-gregorian calendar '
                                      '{} not supported.'.format(rscale))
    else:
        raise ValueError('Invalid RSCALE component: {}'.format(rscale))
```

Note that I also support lower and mixed-case calendar names there with the `upper` call. That said, an alternative to using string comparison at all is to simply have an enum-type class that enumerates the supported calendars. That seems like it might have some advantages here, but depending on the implementation it might make it harder for people to extend `dateutil` to support other calendars. String comparison seems like a good start.